### PR TITLE
[1/N] fix CP for Helios

### PR DESCRIPTION
**Pull request by SHYuanBest**
Under CP, the noise = self.sample_block_noise(...) on the two ranks will sample random noise independently because dist.sample does not synchronize random seeds. This results in different start_point_list[i_s] for the two ranks, and the DMD denoising starting point of scheduler.step will be inconsistent. The two ranks will produce different latents, and ultimately rank 0 and rank 1 will decode different video frames, but only the result of one rank will be actually used (or the two results will be mixed in some incorrect way). Therefore, we need to use a generator to control randomness.

related to https://github.com/huggingface/diffusers/pull/13223

### CODE REVIEW

File: vllm_omni/diffusion/models/helios/pipeline_helios.py [hunk 1]
```diff
@@ -873,18 +873,31 @@ def _stage2_sample(
 
         return latents
 
-    def sample_block_noise(self, batch_size, channel, num_frames, height, width, patch_size=(1, 2, 2)):
+    def sample_block_noise(self, batch_size, channel, num_frames, height, width, patch_size=(1, 2, 2), device="cuda"):
+        # NOTE: A generator must be provided to ensure correct and reproducible results.
+        # Creating a default generator here is a fallback only — without a fixed seed,
+        # the output will be non-deterministic and may produce incorrect results in CP context.
+        generator = torch.Generator(device=device)
```
Comment (gcanlin) [inline vllm_omni/diffusion/models/helios/pipeline_helios.py]: Can we use the same generator in main forward, which can be set by users?

https://github.com/vllm-project/vllm-omni/blob/c3a59f10f3f291f923f6288e269ee373af4568d6/vllm_omni/diffusion/models/helios/pipeline_helios.py#L250

Comment (SHYuanBest) [reply vllm_omni/diffusion/models/helios/pipeline_helios.py]: have modified

File: vllm_omni/diffusion/models/helios/pipeline_helios.py [hunk 2]
```diff
@@ -873,18 +875,32 @@ def _stage2_sample(
 
         return latents
 
-    def sample_block_noise(self, batch_size, channel, num_frames, height, width, patch_size=(1, 2, 2)):
+    def sample_block_noise(self, batch_size, channel, num_frames, height, width, patch_size=(1, 2, 2), device="cuda", generator=None):
+        # NOTE: A generator must be provided to ensure correct and reproducible results.
+        # Creating a default generator here is a fallback only — without a fixed seed,
+        # the output will be non-deterministic and may produce incorrect results in CP context.
+        if generator is None:
```
Comment (hsliuustc0106) [inline vllm_omni/diffusion/models/helios/pipeline_helios.py]: @gcanlin does it work for all platform?

Comment (gcanlin) [reply vllm_omni/diffusion/models/helios/pipeline_helios.py]: Can remove this check and the parameter `device`. We have the defensive check for generator in model runner so this code will never be access.
https://github.com/vllm-project/vllm-omni/blob/c3a59f10f3f291f923f6288e269ee373af4568d6/vllm_omni/diffusion/worker/diffusion_model_runner.py#L209-L216

**Review by gcanlin — APPROVED** [review]
LGTM now, thanks!

**Comment by SamitHuang** [comment]
can you share the test result? @SHYuanBest
